hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -217,6 +217,37 @@ static const struct usage_map xpadneo_usage_maps[] = {
 
 	/* disable duplicate button */
 	USAGE_IGN(0xC0224),
+
+	/* XBE2: Disable "dial", which is a redundant representation of the D-Pad */
+	USAGE_IGN(0x10037),
+
+	/* XBE2: Disable blind axes */
+	USAGE_IGN(0x10040), /* Vx */
+	USAGE_IGN(0x10041), /* Vy */
+	USAGE_IGN(0x10042), /* Vz */
+	USAGE_IGN(0x10043), /* Vbrx */
+	USAGE_IGN(0x10044), /* Vbry */
+	USAGE_IGN(0x10045), /* Vbrz */
+
+	/* XBE2: Disable duplicate buttons */
+	USAGE_IGN(0x90010), /* copy of A */
+	USAGE_IGN(0x90011), /* copy of B */
+	USAGE_IGN(0x90013), /* copy of X */
+	USAGE_IGN(0x90014), /* copy of Y */
+	USAGE_IGN(0x90016), /* copy of LB */
+	USAGE_IGN(0x90017), /* copy of RB */
+	USAGE_IGN(0x9001B), /* copy of Start */
+	USAGE_IGN(0x9001D), /* copy of LS */
+	USAGE_IGN(0x9001E), /* copy of RS */
+	USAGE_IGN(0xC0082), /* copy of Select button */
+
+	/* XBE2: Disable unused "TRIGGER_HAPPY" buttons */
```

Comment:
You may need to map some of these to other symbols than `BTN_TRIGGER_HAPPY_*` when you're going to deal with the paddles, ideally these should get new `0x9....` codes through bit-shifting, directly following the `0x9000b` for the logo button. On a side note: I wonder how much beer went into the kernel development when naming the buttons. :-)

Due to ordering constraints, you can only use button symbols below `#define BTN_DIGI 0x140` (kernel source `input-event-codes.h`) otherwise the input layer will order them in front of `BTN_A` which we don't want (messes up `jstest`).

Reply:
OTOH, there don't seem to be many other options than `BTN_TRIGGER_HAPPY*`.